src/plugins/explore/public/application/utils/state_management/selectors/index.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSelector } from '@reduxjs/toolkit';
+import { RootState } from '../store';
+
+/**
+ * Basic selectors
+ */
+const selectQueryState = (state: RootState) => state.query;
+export const selectUIState = (state: RootState) => state.ui;
+const selectResultsState = (state: RootState) => state.results;
+const selectLegacyState = (state: RootState) => state.legacy;
+export const selectTabState = (state: RootState) => state.tab;
+
+/**
+ * Query selectors
+ */
+export const selectQuery = createSelector([selectQueryState], (queryState) => queryState);
+
+export const selectQueryString = createSelector(
+  [selectQueryState],
+  (queryState) => queryState.query
+);
+
+export const selectQueryLanguage = createSelector(
+  [selectQueryState],
+  (queryState) => queryState.language
+);
+
+export const selectDataset = createSelector([selectQueryState], (queryState) => queryState.dataset);
+
+/**
+ * UI selectors
+ */
+export const selectActiveTabId = createSelector([selectUIState], (uiState) => uiState.activeTabId);
+
+export const selectShowDatasetFields = createSelector(
```

Comment:
showFields

Reply:
is this comment about the name? i personally think just `fields` is too generic maybe?